client/wfnews-war/src/main/angular/src/app/components/full-details/area-restrictions-full-details/area-restrictions-full-details.component.ts

```diff
@@ -128,24 +127,36 @@ export class AreaRestrictionsFullDetailsComponent extends FullDetailsComponent i
           this.fireCentre = response.features[0].attributes.FIRE_CENTRE_NAME + " Fire Centre";
           this.issuedDate = convertToDateYear(response.features[0].attributes.ACCESS_STATUS_EFFECTIVE_DATE);
           this.bulletinUrl = response.features[0].attributes.BULLETIN_URL;
+          this.centroidLatitude = response.features[0].centroid.y;
+          this.centroidLongitude = response.features[0].centroid.x;
           let restrictionPolygon = response.features[0].geometry.rings;
           this.populateIncident(restrictionPolygon)
-          this.latitude = response.features[0].centroid.y;
-          this.longitude = response.features[0].centroid.x;
-          this.initMap(this.latitude, this.longitude);
       }
    })
  }
 
  populateIncident(restrictionPolygon: [][]) {
-  let poly = restrictionPolygon[0]
+  let poly: number [][] = restrictionPolygon[0]
+  let polyArray:Array<number> [] = [];
+ 
+  poly.forEach(value => {
+    polyArray.push(value)
+  })
+  
+  let multiPolyArray = [polyArray];
+  let turf = window['turf']
+  let bufferedPolygon = turf.polygon(multiPolyArray)
+  let buffer = turf.buffer(bufferedPolygon, 10, {
+    units: 'kilometers'
+  });
+
   let xArray: number [] = [];
   let yArray: number [] = [];
 
-  poly.forEach((value) => {
-    xArray.push(value[0]);
-    yArray.push(value[1]);
-  })
+  buffer.geometry.coordinates[0].forEach((value) => {
```

Comment:
Turf has a BBOX function that you can use here. `turf.bbox` can take a polygon or multipoly and calculate the bbox from that

Also, try to avoid forEach where we can. It changes scope and is less efficient than using for of

Reply:
I've update now based on your comments, thanks